src/components/views/messages/MessageActionBar.js

```diff
@@ -87,32 +123,117 @@ export default class MessageActionBar extends React.PureComponent {
         this.onFocusChange(true);
     }
 
-    render() {
+    isContentActionable() {
         const { mxEvent } = this.props;
         const { status: eventStatus } = mxEvent;
 
         // status is SENT before remote-echo, null after
         const isSent = !eventStatus || eventStatus === EventStatus.SENT;
 
-        let replyButton;
-
         if (isSent && mxEvent.getType() === 'm.room.message') {
             const content = mxEvent.getContent();
             if (
                 content.msgtype &&
                 content.msgtype !== 'm.bad.encrypted' &&
                 content.hasOwnProperty('body')
             ) {
-                replyButton = <span className="mx_MessageActionBar_replyButton"
-                    title={_t("Reply")}
-                    onClick={this.onReplyClick}
-                />;
+                return true;
             }
         }
 
+        return false;
+    }
+
+    isReactionsEnabled() {
+        return SettingsStore.isFeatureEnabled("feature_reactions");
+    }
+
+    renderAgreeDimension() {
+        if (!this.isReactionsEnabled()) {
+            return null;
+        }
+
+        const state = this.state.agreeDimension;
+        const options = [
+            {
+                key: "agree",
+                content: "👍",
+                onClick: this.onAgreeClick,
+            },
+            {
+                key: "disagree",
+                content: "👎",
+                onClick: this.onDisagreeClick,
```

Comment:
why not use the dimension here and bind an onClick to toggleDimensionValue instead of having several pointer functions?

Something like:
```
{
  key: "disagree",
  content: "👎",
  dimension: "agreeDimension",
}
```
then in `renderReactionDimensionItems`:
```
<span ... onClick={this.toggleDimensionValue.bind(this, option.dimension, option.key)}> ...
```

Reply:
I thought about that, but it's [generally suggested](https://reactjs.org/docs/handling-events.html) that you avoid using inline callbacks or bind in React render methods, since would create a new function every time, and if that were passed to lower components, it could trigger unnecessary re-rendering.

In this case, we are just rendering `span`s, not React components for now, so we wouldn't have this issue, but I tend to just avoid doing it everywhere with React, especially since you might later extract some code to a component and not think about this issue.

I think if `MessageActionBar` gets too complex and/or these functions feel like too much noise, then I think pulling out the reaction dimensions to a new component as you suggested is the correct thing to do.